CDCgov/prime-reportstream
# 12554-Move-creation-of-ObjectMapper-to-JacksonMapperUtilities

This PR:

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Applies singleton pattern to the usage of jackson object mapper.

Test Steps:
1. Checkout PR to local repo
2. Run and pass all tests

Profiled (gradle profiler) the end2end_up PR vs master, make sure no negative impact on latencies:

PR elapsed time avg:

![image](https://github.com/CDCgov/prime-reportstream/assets/135686833/51cfae03-4fbf-48a2-9b5c-84fe26b34d34)

Master elapsed time avg:

![image](https://github.com/CDCgov/prime-reportstream/assets/135686833/85e091a4-b9f3-4642-aac2-d73dd5e656ea)


 
## Changes
- Added singleton object mappers of flavors into JacksonMapperUtilities
- Refactored code base - main code and tests where direct ObjectMapper / jsonObjectMapper constructors used, replace with the singleton usage in JacksonMapperUtilities.

## Checklist


### Testing

### Process

## Linked Issues
- Fixes #12554

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #12800

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
